### PR TITLE
code coverage initiative

### PR DESCRIPTION
**Pull request by kunkun-tang**
Code coverage measures what percentage of code has been exercised by test suites. In order to honor the measurement, This PR adds codecov plugin into travis CI. The detailed documentation of this plugin can be found: https://docs.codecov.io/docs

A Sample how code coverage works for individual commit: https://codecov.io/gh/kunkun-tang/azkaban/commits

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -227,3 +231,34 @@ if (System.env.TRAVIS == 'true') {
...
 }
+
+def publishedProjects = subprojects.findAll()
+ 
+task jacocoRootReport(type: JacocoReport, group: 'Coverage reports') {
+    description = 'Generates an aggregate report from all subprojects'
+ 
+    dependsOn(publishedProjects.test)
+ 
+    additionalSourceDirs = files(publishedProjects.sourceSets.main.allSource.srcDirs)
```
Comment (HappyRay) [inline build.gradle]: It is not clear to me what these lines are for.

Could you add some comments? e.g. reference online documentations?

**Comment by HappyRay** [comment]
It's great to see the code coverage report.

**Comment by kunkun-tang** [comment]
Good ask. @jamiesjc 
Created a task to track. https://github.com/azkaban/azkaban/issues/1843
In fact, I'm about to discard the currrent documentation page and propose the new ones alternative. Will do it there.

**Comment by HappyRay** [comment]
Is there a test coverage badge that we can add to the project like the travis badge we have?

**Pull request merged**